config/src/grt.rs

```diff
@@ -4,10 +4,18 @@
 use bigdecimal::{BigDecimal, ToPrimitive};
 use serde::{de::Error, Deserialize};
 
-#[derive(Debug, PartialEq)]
+#[derive(Debug, PartialEq, Default)]
 pub struct NonZeroGRT(u128);
 
 impl NonZeroGRT {
+    pub fn new(value: u128) -> Result<Self, String> {
+        if value == 0 {
```

Comment:
Have you ever used something like https://doc.rust-lang.org/stable/std/num/type.NonZeroU128.html for this?

Reply:
TIL, but this is used only in tests.